src/doc/nomicon/vec-final.md

```diff
@@ -226,7 +226,11 @@ impl<T> Iterator for RawValIter<T> {
         } else {
             unsafe {
                 let result = ptr::read(self.start);
-                self.start = self.start.offset(1);
+                self.start = if mem::size_of::<T>() == 0 {
+					(self.start as usize + 1) as *const _
```

Comment:
It looks like you have literal tabs in the file, you should indent with only spaces.